WifiART/src/userinterface/ArtMvcView.java

```diff
@@ -174,90 +189,355 @@ private void setupOutputFileChooserPanel(JPanel panel) {
+    /**
+     * Get whether or not the LCI subelement is to be included in the output
+     *
+     * @return whether or not the LCI subelement is included.
+     */
     public boolean getLciIncluded() {
         return lciCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Z subelement is to be included in the output
+     *
+     * @return whether or not the Z subelement is included.
+     */
     public boolean getZIncluded() {
         return zCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Usage Rules/Policy subelement is to be included in the output
+     *
+     * @return whether or not the Usage Rules/Policy subelement is included.
+     */
     public boolean getUsageIncluded() {
         return usageCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the BSSID List subelement is to be included in the output
+     *
+     * @return whether or not the BSSID List subelement is included.
+     */
     public boolean getBssidIncluded() {
         return bssidCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Location Civic subelement is to be included in the output
+     *
+     * @return whether or not the Location Civic subelement is included.
+     */
     public boolean getLcrIncluded() {
         return lcrCheckbox.isSelected();
     }
+
+    /**
+     * Get whether or not the Map Image subelement is to be included in the output
+     *
+     * @return whether or not the Map Image subelement is included.
+     */
     public boolean getMapIncluded() {
         return mapCheckbox.isSelected();
     }
 
+    /**
+     * Get the name of the input file.
+     *
+     * @return the name of the input file
+     */
     public String getInputFileName() {
         return inputFileNameField.getText();
     }
+
+    /**
+     * Get the input file directory.
+     *
+     * @return the input file directory
+     */
     public String getInputDir() {
         return inputDirField.getText();
     }
+
+    /**
+     * Get the name of the output file.
+     *
+     * @return the name of the output file
+     */
     public String getOutputFileName() {
         return outputFileNameField.getText();
     }
+
+    /**
+     * Get the output file directory.
+     *
+     * @return the output file directory
+     */
     public String getOutputDir() {
         return outputDirField.getText();
     }
 
-    public void setLciIncludedListener(ActionListener listener) {
+    /**
+     * Get whether or not the buffer should be displayed in a readable format.
+     *
+     * @return whether or not the buffer should be displayed in a readable format.
+     */
+    public boolean getReadable() {
+        return readableBufferCheckbox.isSelected();
+    }
+
+    /**
+     * Get whether or not the encoding should match the new Android version.
+     *
+     * @return whether or not the encoding should match the new Android version.
+     */
+    public boolean isAndroidVersionNew() {
```

Comment:
"new" is too ambiguous and will become outdated soon. Maybe isAndroidVersionAtLeastS()?

Reply:
Good idea. That is much more precise.